lib/storage/merge.go

```diff
@@ -45,7 +45,25 @@ func mergeBlockStreamsInternal(ph *partHeader, bsw *blockStreamWriter, bsm *bloc
 	defer putBlock(pendingBlock)
 	tmpBlock := getBlock()
 	defer putBlock(tmpBlock)
-	for bsm.NextBlock() {
+
+	var localRowsMerged, localRowsDeleted atomic.Uint64
+	updateStats := func() {
```

Comment:
This is a great idea, which should go into a separate pull request, since the rest of this pull request in't so good (it doesn't resolve reliably the false sharing issue across global variables, while it increases code complexity in unobvious places).

I implemented this idea in the commit 23fd269ccf07183a29d8d3002c07f44431664895 . This commit also applies the same technique of periodic updating the global stats during background merges at `lib/mergeset`.